compiler/rustc_borrowck/src/region_infer/opaque_types.rs

```diff
@@ -18,6 +19,47 @@ use crate::session_diagnostics::{LifetimeMismatchOpaqueParam, NonGenericOpaqueTy
 use crate::universal_regions::RegionClassification;
 
 impl<'tcx> RegionInferenceContext<'tcx> {
+    pub(crate) fn propagate_opaque_types(
+        &self,
+        infcx: &InferCtxt<'tcx>,
+        opaque_ty_decls: FxIndexMap<OpaqueTypeKey<'tcx>, OpaqueHiddenType<'tcx>>,
+    ) -> Vec<InClosureRequirement<(OpaqueTypeKey<'tcx>, OpaqueHiddenType<'tcx>)>> {
+        let mut propagated_decls = Vec::new();
+        for (opaque_type_key, concrete_type) in opaque_ty_decls {
+            let mut guar = None;
+            let propagated = fold_regions(infcx.tcx, (opaque_type_key, concrete_type), |r, _| {
```

Comment:
current status
```rust
fn define_in_closure<'a>() -> impl Sized + 'a {
    let _ = || {
        let _: () = define_in_closure::<'a>();
    };
}
```
fails to propagate the opaque as it's `opaque<'unconstrained, 'a>()` need to explicitly handle uncaptured args here :skull:

Reply:
If we enforce WF for the propagated opaque, shouldn't that constrain `'unconstrained = 'a`?

Reply:
I guess it doesn't help for truly uncaptured opaque lifetimes tho.